### PR TITLE
[WIP] Update Readme.md Flow

### PR DESCRIPTION
**Pull request by amarjandu**
Address issues of readability and provide inside to deploying infra for DSS
Sort `doc/environment/readme.md` for easier locating of values.
Add Table of Contents to `readme.md` for easier access
Reorganize sections in `readme.md`

Resolves: #1960

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -232,12 +306,37 @@ And you should be able to list bundles like this:
...
+principal to that group. Because this is a limited set of permissions, it does not have write access to IAM. To set up
+the IAM policies for resources in your account that the app will use, run `make deploy` using privileged account
+credentials once from your workstation. After this is done, Travis CI will be able to deploy on its own. You must
+repeat the `make deploy` step from a privileged account any time you change the IAM policies templates in
+`iam/policy-templates/`.
+
+### Authorizing the event relay
+
+Environment variables provide the AWS credentials needed to relay events originating from supported cloud platforms
+outside of AWS. Run `scripts/create_config_aws_event_relay_user.py` to create an AWS IAM user with the appropriate
```
[the remark this answers is not in the document]
Comment (amarjandu) [reply README.md]: Yeah i just experienced this within my deployment today, ill note it down thanks

File: README.md [hunk 2]
```diff
@@ -210,15 +260,39 @@ incorporate dependent infrastructure, such SQS queues or SNS topics, by placing
...
 
+##### Resources
+Cloud resources have the potential for naming collision in both [AWS](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html)
+ and [GCP](https://cloud.google.com/storage/docs/naming), ensure to rename resources as needed.
+
+#### Buckets
+
+Buckets within AWS and GCP need to be available for use by the DSS. Use Terraform to setup these resources:
+```
+make -C infra COMPONENT=buckets apply
```
[the remark this answers is not in the document]
Comment (amarjandu) [reply README.md]: !!!!! thanks almost ruined deployments! changed

File: README.md [hunk 3]
```diff
@@ -210,15 +260,39 @@ incorporate dependent infrastructure, such SQS queues or SNS topics, by placing
 usual plan/review Terraform workflow, and should therefore be lightweight in nature. Large infrastructure should be
 added to `$DSS_HOME/infra` instead.
 
+##### Resources
+Cloud resources have the potential for naming collision in both [AWS](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html)
```
[the remark this answers is not in the document]
Comment (amarjandu) [reply README.md]: resolved

File: README.md [hunk 4]
```diff
@@ -210,15 +260,39 @@ incorporate dependent infrastructure, such SQS queues or SNS topics, by placing
...
+ and [GCP](https://cloud.google.com/storage/docs/naming), ensure to rename resources as needed.
```
[the remark this answers is not in the document]
Comment (amarjandu) [reply README.md]: resolved

File: README.md [hunk 5]
```diff
@@ -210,15 +260,39 @@ incorporate dependent infrastructure, such SQS queues or SNS topics, by placing
...
+
+#### Buckets
+
+Buckets within AWS and GCP need to be available for use by the DSS. Use Terraform to setup these resources:
+```
+make -C infra COMPONENT=buckets apply
+```
+
+#### ElasticSearch
 The AWS Elasticsearch Service is used for metadata indexing. For typical development deployments the
```
[the remark this answers is not in the document]
Comment (amarjandu) [reply README.md]: terraform has 5.5 hard-coded, using that as version info.

File: README.md [hunk 6]
```diff
@@ -210,15 +260,39 @@ incorporate dependent infrastructure, such SQS queues or SNS topics, by placing
...
+t2.small.elasticsearch instance type is sufficient. Use the [`DSS_ES_`](./docs/environment/README.md) variables to adjust the cluster as needed. 
 
+Add allowed IPs for ElasticSearch to the secret manager, use comma separated IPs:
+```
+echo ' ' | ./scripts/set_secret.py --secret-name $ES_ALLOWED_SOURCE_IP_SECRETS_NAME
+
+```
+Use Terraform to deploy ES resource:
+```
+make -C infra COMPONENT=elasticsearch apply
```
[the remark this answers is not in the document]
Comment (amarjandu) [reply README.md]: addressed.

**Comment by amarjandu** [comment]
@DailyDreaming i added your changes requested, im leaving the comment threads open to allow @natanlao  to read them over, and hopefully give insight to what you saw.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (DailyDreaming) [reply README.md]: Approving.  Please add this in before merging.

**Pull request merged**